examples/app-with-extensions/src/module.tsx

```diff
@@ -43,17 +43,33 @@ export const plugin = new AppPlugin<{}>()
     description: 'This link will only be visible on time series and pie charts',
     extensionPointId: PluginExtensionPoints.DashboardPanelMenu,
     onClick: (_, { openModal, context }) => {
-      openModal({
-        title: 'Modal opened from onClick',
-        body: () => <Modal panelTitle={context?.title} />,
-      });
+      const targets = context?.targets;
+      const title = context?.title;
+
+      if (!Array.isArray(targets) || targets.length === 0) {
+        return;
+      }
+
+      if (targets.length > 1) {
```

Comment:
Maybe add a comment above this block to explain our intention?

```typescript
// Show a modal to display a UI for selecting between the available queries (targets) 
// in case there are more available. 
if (targets.length > 1) {
```